kong/tracing/request_id.lua

```diff
@@ -0,0 +1,50 @@
+local base = require "resty.core.base"
+
+local get_request = base.get_request
+
+local NGX_VAR_PHASES = {
+  set           = true,
+  rewrite       = true,
+  access        = true,
+  content       = true,
+  header_filter = true,
+  body_filter   = true,
+  log           = true,
+  balancer      = true,
+}
+
+
+local function get_ctx_request_id()
+  return ngx.ctx.request_id
+end
+
+
+local function get()
+  if not get_request() then
```

Comment:
We don't need to use `get_request` here, `ngx.get_phase` would return `init` in that case and we can fall to the below branch:

https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/phase.lua#L41-L43

This condition should basically never happen in the actual use case.

Reply:
done, thanks